### PR TITLE
[WIP] Rename v2 API to v3

### PR DESCRIPTION
**Pull request by robbrockbank**
## Description

Renaming all of our v2 APIs to be v3 instead.

I've done this as two separate commits.
-   The first is a blanket replace of v2 -> v3
-   The second is undoing changes that were not required

(this way you can see what i've explicitly excluded).

## Todos
- [x] Tests
- [ ] Documentation
- [ ] Release note
- [ ] Update every other repo under the sun

## Release Note

```release-note
None required
```

### CODE REVIEW

File: lib/backend/k8s/conversion/conversion_test.go [hunk 1]
```diff
@@ -1095,7 +1095,7 @@ var _ = Describe("Test NetworkPolicy conversion (k8s <= 1.7, no policyTypes)", f
 								PodSelector: &metav1.LabelSelector{
 									MatchLabels: map[string]string{
 										"k":  "v",
-										"k2": "v2",
+										"k2": "v3",
```
Comment (robbrockbank) [inline lib/backend/k8s/conversion/conversion_test.go]: didn't really need to change this - but more hassle that it's worth to undo just this.

File: lib/backend/compat/compat.go [hunk 2]
```diff
@@ -732,7 +732,7 @@ func toDatastoreGlobalBGPConfigKey(key model.GlobalBGPConfigKey) model.GlobalBGP
 }
 
 // toDatastoreGlobalBGPConfigList modifies the Global BGP Config List interface to the one required by
-// the datastore (for back-compatibility with what is expected in teh etcdv2 datastore driver).
+// the datastore (for back-compatibility with what is expected in teh etcdv3 datastore driver).
```
Comment (bcreane) [inline lib/backend/compat/compat.go]: spotted an old typo: `teh`

File: lib/backend/k8s/conversion/conversion_test.go [hunk 3]
```diff
@@ -272,7 +272,7 @@ var _ = Describe("Test NetworkPolicy conversion", func() {
...
```
Comment (bcreane) [inline lib/backend/k8s/conversion/conversion_test.go]: this looks questionable

File: lib/backend/k8s/conversion/conversion_test.go [hunk 4]
```diff
@@ -435,7 +435,7 @@ var _ = Describe("Test NetworkPolicy conversion", func() {
 							extensions.NetworkPolicyPeer{
 								PodSelector: &metav1.LabelSelector{
 									MatchLabels: map[string]string{
-										"k2": "v2",
+										"k2": "v3",
```
Comment (bcreane) [inline lib/backend/k8s/conversion/conversion_test.go]: not sure this is right

File: lib/backend/k8s/conversion/conversion_test.go [hunk 5]
```diff
@@ -515,7 +515,7 @@ var _ = Describe("Test NetworkPolicy conversion", func() {
...
```
Comment (bcreane) [inline lib/backend/k8s/conversion/conversion_test.go]: same question here ..

File: lib/backend/k8s/conversion/conversion_test.go [hunk 6]
```diff
@@ -708,7 +708,7 @@ var _ = Describe("Test NetworkPolicy conversion", func() {
 						metav1.LabelSelectorRequirement{
 							Key:      "k",
 							Operator: metav1.LabelSelectorOpIn,
-							Values:   []string{"v1", "v2"},
+							Values:   []string{"v1", "v3"},
```
Comment (bcreane) [inline lib/backend/k8s/conversion/conversion_test.go]: and here ...

File: lib/backend/k8s/conversion/conversion_test.go (hunk 1 above)
Comment (bcreane) [reply lib/backend/k8s/conversion/conversion_test.go]: ah okay, you can disregard previous comments on spurious `LabelSelector` changes

File: lib/backend/k8s/k8s_test.go [hunk 7]
```diff
@@ -39,27 +39,27 @@ import (
 
 var (
 	zeroOrder                  = float64(0.0)
-	calicoAllowPolicyModelSpec = capiv2.GlobalNetworkPolicySpec{
+	calicoAllowPolicyModelSpec = capiv3.GlobalNetworkPolicySpec{
```
Comment (bcreane) [inline lib/backend/k8s/k8s_test.go]: ideally `capiv2` becomes `apiv3`

File: lib/hwm/hwm.go [hunk 8]
```diff
@@ -17,7 +17,7 @@ package hwm
 
 import (
 	log "github.com/sirupsen/logrus"
-	"gopkg.in/tchap/go-patricia.v2/patricia"
+	"gopkg.in/tchap/go-patricia.v3/patricia"
```
Comment (bcreane) [inline lib/hwm/hwm.go]: this looks wrong

File: lib/backend/compat/compat.go (hunk 2 above)
Comment (robbrockbank) [reply lib/backend/compat/compat.go]: yeah saw that and thought meh.  I've actually just reverted that file.  We can probably bin the combat module completely - but wanted to keep it around until I'd done the migration code.

File: lib/backend/k8s/conversion/conversion_test.go (hunk 5 above)
Comment (robbrockbank) [reply lib/backend/k8s/conversion/conversion_test.go]: yeah I think you saw my comment.  This change wasn't necessary but it doesn't actually make any difference - we're just testing a different label value.

Comment (robbrockbank) [reply lib/backend/k8s/conversion/conversion_test.go]: i'll change.  it's beginning to bug me now.

**Pull request merged**